mr/src/main/java/org/apache/iceberg/mr/Catalogs.java

```diff
@@ -180,47 +189,87 @@ public static boolean dropTable(Configuration conf, Properties props) {
   /**
    * Returns true if HiveCatalog is used
    * @param conf a Hadoop conf
+   * @param props the controlling properties
    * @return true if the Catalog is HiveCatalog
    */
-  public static boolean hiveCatalog(Configuration conf) {
-    return HIVE.equalsIgnoreCase(conf.get(InputFormatConfig.CATALOG));
+  public static boolean hiveCatalog(Configuration conf, Properties props) {
+    String catalogName = props.getProperty(InputFormatConfig.TABLE_CATALOG);
+    return HIVE.equalsIgnoreCase(getCatalogType(conf, catalogName));
```

Comment:
Does this handle the default catalog correctly?

Reply:
If the catalogName is not among the table properties, the getCatalogType() will be called with a null name. Inside the getCatalogType() I have the following value assignment `String name = catalogName == null ? DEFAULT_CATALOG : catalogName;`